Mathlib/Topology/Compactness/Lindelof.lean

```diff
@@ -0,0 +1,357 @@
+  simpa only [disjoint_comm] using hs.disjoint_nhdsSet_left
+
+/-- For every family of closed sets whose intersection avoids a Lindelö set,
+there exists a countable subfamily whose intersection avoids this Lindelöf set. -/
+theorem IsLindelof.elim_countable_subfamily_closed {ι : Type v} (hs : IsLindelof s)
+    (t : ι → Set X) (htc : ∀ i, IsClosed (t i)) (hst : (s ∩ ⋂ i, t i) = ∅) :
+    ∃ u : Set ι, u.Countable ∧ (s ∩ ⋂ i ∈ u, t i) = ∅ := by
+    let U := tᶜ
+    have hUo : ∀ i, IsOpen (U i) := by simp only [Pi.compl_apply, isOpen_compl_iff]; exact htc
+    have hsU : s ⊆ ⋃ i, U i := by
+      simp only [Pi.compl_apply]
+      rw [← compl_iInter]
+      apply disjoint_compl_left_iff_subset.mp
+      simp only [compl_iInter, compl_iUnion, compl_compl]
+      apply Disjoint.symm
+      exact disjoint_iff_inter_eq_empty.mpr hst
+    rcases hs.elim_countable_subcover U hUo hsU with ⟨u, ⟨hucount, husub⟩⟩
+    use u, hucount
+    rw [← disjoint_compl_left_iff_subset] at husub
+    simp only [Pi.compl_apply, compl_iUnion, compl_compl] at husub
+    exact disjoint_iff_inter_eq_empty.mp (Disjoint.symm husub)
+
+/--To show that a Lindelöf set intersects the intersection of a family of closed sets,
+  it is sufficient to show that it intersects every countable subfamily. -/
+theorem IsLindelof.inter_iInter_nonempty {ι : Type v} (hs : IsLindelof s) (t : ι → Set X)
+    (htc : ∀ i, IsClosed (t i)) (hst : ∀ u : Set ι, u.Countable ∧ (s ∩ ⋂ i ∈ u, t i).Nonempty) :
+    (s ∩ ⋂ i, t i).Nonempty := by
+  contrapose! hst
+  rcases hs.elim_countable_subfamily_closed t htc hst with ⟨u, ⟨_, husub⟩⟩
+  use u
+  apply fun _ ↦ husub
+
+/-- For every open cover of a Lindelöf set, there exists a countable subcover. -/
+theorem IsLindelof.elim_countable_subcover_image {b : Set ι} {c : ι → Set X} (hs : IsLindelof s)
+    (hc₁ : ∀ i ∈ b, IsOpen (c i)) (hc₂ : s ⊆ ⋃ i ∈ b, c i) :
+    ∃ b', b' ⊆ b ∧ Set.Countable b' ∧ s ⊆ ⋃ i ∈ b', c i := by
+  simp only [Subtype.forall', biUnion_eq_iUnion] at hc₁ hc₂
+  rcases hs.elim_countable_subcover (fun i ↦ c i : b → Set X) hc₁ hc₂ with ⟨d, hd⟩
+  refine ⟨Subtype.val '' d, ?_, Countable.image hd.1 Subtype.val, ?_⟩-- d.image _, _⟩
+  · simp
+  · rw [biUnion_image]
+    apply hd.2
+
+
+/-- A set `s` is Lindelöf if for every open cover of `s`, there exists a countable subcover. -/
+theorem isLindelof_of_countable_subcover
+    (h : ∀ {ι : Type u} (U : ι → Set X), (∀ i, IsOpen (U i)) → (s ⊆ ⋃ i, U i) →
+    ∃ t : Set ι, t.Countable ∧ s ⊆ ⋃ i ∈ t, U i) :
+    IsLindelof s := fun f hf hfs ↦ by
+  contrapose! h
+  simp only [ClusterPt, not_neBot, ← disjoint_iff, SetCoe.forall',
+    (nhds_basis_opens _).disjoint_iff_left] at h
+  choose fsub U hU hUf using h
+  refine ⟨s, U, fun x ↦ (hU x).2, fun x hx ↦ mem_iUnion.2 ⟨⟨x, hx⟩, (hU _).1 ⟩, ?_ ⟩
+  intro t ht h
+  have uinf := f.sets_of_superset (le_principal_iff.1 fsub) h
+  have uninf : ⋂ i ∈ t, (U i)ᶜ ∈ f := by refine (countable_bInter_mem ht).mpr (fun _ _ ↦ hUf _)
```

Comment:
As a closing tactic, `exact` is preferred over `refine` or `apply`. (In this case, you can just remove the `by refine`, just giving a proof term.)
You have at least one other such place.